lib/stats-query-commands.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2017 Balabit
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "stats-query-commands.h"
+#include "stats/stats-query.h"
+#include "messages.h"
+
+enum QueryCommands
+{
+  QUERY_GET = 0,
+  QUERY_GET_AGGREGATE,
+  QUERY_LIST,
+  QUERY_CMD_MAX
+};
+
+typedef GString *(*QUERY_CMD)(const gchar *hds_path);
+
```

Comment:
Please don't use ALL_CAPS for type names.
What is "hds_path"? I don't get the meaning of "hds", maybe "stats_path"? Is that the right term?